### PR TITLE
[CLI] fix u64 to u256 input args

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

This fix the issue #8982  when parsing u64, u128 and u256 args from the CLI. This was originally broken by the nested vector PR #7790  If we look at the changes carefully, we were explicitly casting to string `.to_string()` for u62 - u256, on the `.to_json()` method, but the nested vector PR missed this part.

### Test Plan

**All tested on devnet**

Created a move func and tested make sure u64 works
```
➜  hello_blockchain git:(main) aptos-debug move view --function-id default::message::test_u64 --args u64:1
{
  "Result": [
    "1"
  ]
}

➜  devnet git:(jin/fix_u64_u128_u256_args) ✗ aptos-debug move view --function-id \
devnet::cli_e2e_tests::test_big_number \
--args \
u64:123 \  
u128:12345 \ 
u256:340282366920938463463374607431768211455 
{
  "Result": [
    "123"
  ]
}

```

Make sure nested u64 on using "move run" still works
```
➜  e2e git:(main) ✗ aptos move run \
--function-id default::cli_args::test_nested_u64 \
--args \
'u64:[[1, 2], [3], [4]]'
Do you want to submit a transaction for a range of [200 - 300] Octas at a gas unit price of 100 Octas? [yes/no] >
yes
{
  "Result": {
    "transaction_hash": "0xc323b96a8d4585ce0e4c7e62ccab8b2adec8bdd55cf682db1dfc31ef47e4e4ca",
    "gas_used": 2,
    "gas_unit_price": 100,
    "sender": "e6fc96a9464ef0a85fd7146fdbae0a48c5930e44d5552ebca90ad0d0a2c7636c",
    "sequence_number": 9,
    "success": true,
    "timestamp_us": 1689526892222664,
    "version": 17439203,
    "vm_status": "Executed successfully"
  }
}
```

Unit Test
```
2023-07-18 09:53:21,086 - INFO - These tests passed:
2023-07-18 09:53:21,086 - INFO - test_metrics_accessible
2023-07-18 09:53:21,086 - INFO - test_init
2023-07-18 09:53:21,086 - INFO - test_config_show_profiles
2023-07-18 09:53:21,086 - INFO - test_account_fund_with_faucet
2023-07-18 09:53:21,086 - INFO - test_account_create
2023-07-18 09:53:21,086 - INFO - test_account_lookup_address
2023-07-18 09:53:21,086 - INFO - test_aptos_header_included
2023-07-18 09:53:21,086 - INFO - test_move_compile
2023-07-18 09:53:21,086 - INFO - test_move_compile_script
2023-07-18 09:53:21,086 - INFO - test_move_publish
2023-07-18 09:53:21,086 - INFO - test_move_run
2023-07-18 09:53:21,086 - INFO - test_move_view
2023-07-18 09:53:21,086 - INFO - All tests passed!
```

### CODE REVIEW

**Review by davidiw — APPROVED** [review]
would have been great to have shown a test where it failed and now works.

File: crates/aptos/e2e/cases/move.py [hunk 1]
```diff
@@ -200,3 +200,23 @@ def test_move_view(run_helper: RunHelper, test_name=None):
     response = json.loads(response.stdout)
     if response["Result"] == None or response["Result"][0] != True:
         raise TestError("View function did not return correct result")
+    
+    # Test view function with with u64 argument
+    expected_u64 = 123
```
Comment (gregnazario) [inline crates/aptos/e2e/cases/move.py:205]: Can we test with max u64 / u128 / u256?

File: aptos-move/move-examples/cli-e2e-tests/common/sources/cli_e2e_tests.move [hunk 2]
```diff
@@ -257,6 +257,12 @@ module addr::cli_e2e_tests {
         }
     }
 
+    // The following functions are used to test input args
+    #[view]
+    public fun test_u64(n: u64): u64 {
+        n
+    }
+
```
Comment (gregnazario) [inline aptos-move/move-examples/cli-e2e-tests/common/sources/cli_e2e_tests.move:265]: Can we test additionally `[u64, u64]`?

**Review by gregnazario — COMMENTED** [review]
Let's add some more testing, this is a space where insufficient testing kinda led to these kinds of regressions.

Let's add it around the big big numbers and for vectors

**Comment by 0xjinn** [comment]
> Let's add some more testing, this is a space where insufficient testing kinda led to these kinds of regressions.
> 
> Let's add it around the big big numbers and for vectors

Yep!, so this PR only fixes the issues on single level input args. I will have another PR up for the big number issue

**Comment by 0xjinn** [comment]
Updated with MAX u64, u128, and u256. And ensure these big numbers works when running view function.

**Review by gregnazario — APPROVED** [review]

File: crates/aptos/src/move_tool/mod.rs [hunk 3]
```diff
@@ -1513,8 +1513,24 @@ impl ArgWithType {
...
+                    serde_json::to_value(bcs::from_bytes::<U256>(&self.arg)?.to_string())
+                        .map_err(|err| CliError::UnexpectedError(err.to_string()))
+                },
+                FunctionArgType::Raw => serde_json::to_value(&self.arg)
+                    .map_err(|err| CliError::UnexpectedError(err.to_string())),
+                _ => serde_json::to_value(bcs::from_bytes::<T>(&self.arg)?)
+                    .map_err(|err| CliError::UnexpectedError(err.to_string())),
+            },
             1 => serde_json::to_value(bcs::from_bytes::<Vec<T>>(&self.arg)?)
                 .map_err(|err| CliError::UnexpectedError(err.to_string())),
```
[the remark this answers is not in the document]
Comment (0xjinn) [reply crates/aptos/src/move_tool/mod.rs:1535]: Adding the same changes to other depth didn't solve the issue for. I will have another PR to fix this part.

**Comment by 0xjinn** [comment]
Updated `api/goldens/aptos_api__tests__invalid_post_request_test__test_invalid_entry_function_argument_address_type.json `expected text

there is a smoke test failed when I enabled the precision feature on `serde_json`.

**Pull request merged**